src/main/java/org/web3j/tx/ManagedTransaction.java

```diff
@@ -75,8 +81,13 @@ protected TransactionReceipt processResponse(EthSendTransaction transactionRespo
         }
 
         String transactionHash = transactionResponse.getTransactionHash();
-
-        return waitForTransactionReceipt(transactionHash);
+        if (attempts > 0) {
```

Comment:
exactly I'm doing now, please wait until my next PR, probably in a couple of hours